src/Session/Session.php

```diff
@@ -2,108 +2,98 @@
     protected $data;
 
     /**
-     * Instantiates object, sets session data
+     * Instantiates object, sets session data, instantiates a SavedTokens instance
+     *
+     * @param FileCache $file_cache A file cache object
      */
-    public function __construct($fileCache)
+    public function __construct($file_cache)
     {
-        $this->cache = $fileCache;
-        $session = $this->cache->getData('session');
-        $this->data = $session;
-        if (empty($session)) {
-            $this->data = new \stdClass();
-        }
+        $this->cache = $file_cache;
+        $this->data = (object)$this->cache->getData('session');
+        $this->tokens = new SavedTokens(['session' => $this,]);
     }
 
     /**
      * Removes the session from the cache
-     *
-     * @return void
      */
     public function destroy()
     {
         $this->cache->remove('session');
+        $this->data = (object)[];
     }
 
     /**
      * Returns given data property or default if DNE.
      *
      * @param string $key Name of property to return
-     * @param mixed $default Default return value in case property DNE
      * @return mixed
+     * @throws TerminusException If the given key is not located
      */
-    public function get($key, $default = false)
+    public function get($key)
     {
-        if (isset($this->data) && isset($this->data->$key)) {
+        if (isset($this->data->$key)) {
             return $this->data->$key;
         }
-        return $default;
+        throw new TerminusException('The {key} property cannot be found in the cache data.', compact('key'));
     }
 
     /**
-     * Sets a keyed value to be part of the data property object
-     *
-     * TODO: This is never used and arguably shouldn't be (since it doesn't save
-     *       data to the persistent session store)
-     *       This should be removed in favor of setData which is designed to
-     *       instantiate the entire session or changed to add persistent data.
+     * Returns a user with the current session user id
      *
-     * @param string $key Name of data property
-     * @param mixed $value Value of property to set
-     * @return Session
+     * @return User A User object reperesenting the logged-in user
      */
-    public function set($key, $value = null)
+    public function getUser()
     {
-        $this->data->$key = $value;
-        return $this;
+        // TODO: Remove this direct instantiation to make this more testable
+        return new User((object)['id' => $this->get('user_id'),]);
     }
 
     /**
-     * Saves session data to cache
-     *
-     * TODO: The difference between set and setData is confusing. This should be
-     *       refactored when Auth() is.
+     * Responds with the status of this session (i.e. whether the client is logged in)
      *
-     * @param array $data Session data to save
-     * @return bool
+     * @return boolean
      */
-    public function setData($data)
+    public function isActive()
```

Comment:
isAuthenticated sounds more appropriate to me, too. I'll make that change, push, wait for CI, yadda yadda yadda.